addons/mail/static/src/models/welcome_view/welcome_view.js

```diff
@@ -0,0 +1,163 @@
+        // Public
+        //----------------------------------------------------------------------
+
+        /**
+         * @static
+         * @param {string} name 
+         */
+        static async updateGuestNameServerSide(name) {
+            await this.env.services.rpc({
+                route: '/mail/guest/update_name',
+                params: { name },
+            });
+        }
+
+        /**
+         * @param {MouseEvent} ev 
+         */
+        async onClickJoinButton(ev) {
+            this.updateGuestNameIfAnyThenJoinChannel();
+        }
+
+        /**
+         * Handles OWL update on this WelcomeView component.
+         */
+        onComponentUpdate() {
+            this._handleFocus();
+        }
+
+        /**
+         * @param {KeyboardEvent} ev 
+         */
+        onInputGuestNameInput(ev) {
+            this._updateGuestNameWithInputValue();
+        }
+
+        /**
+         * @param {KeyboardEvent} ev 
+         */
+        onKeydownGuestNameInput(ev) {
+            if (ev.key === 'Enter') {
+                this.updateGuestNameIfAnyThenJoinChannel();
+            }
+        }
+
+        /**
+         * Redirects users to the related channel. If the current user is a
+         * guest, first updates their username with the value of guestNameInput.
+         */
+        async updateGuestNameIfAnyThenJoinChannel() {
+            if (this.messaging.currentGuest) {
+                await this.messaging.models['mail.welcome_view'].updateGuestNameServerSide(this.guestName.trim());
+            }
+            window.location = `/discuss/channel/${this.channelId}`;
+        }
+
+        //----------------------------------------------------------------------
+        // Private
+        //----------------------------------------------------------------------
+
+        /**
+         * @private
+         */
+        _handleFocus() {
+            if (this.isDoFocusGuestNameInput) {
+                if (!this.guestNameInputRef.el) {
+                    return
+                }
+                this.update({ isDoFocusGuestNameInput: false });
+                this.guestNameInputRef.el.focus();
+                // place cursor at end of text
+                this.guestNameInputRef.el.value = "";
+                this.guestNameInputRef.el.value = this.guestName;
+            }
+        }
+
+        /**
+         * @private
+         */
+        _updateGuestNameWithInputValue() {
+            this.update({ guestName: this.guestNameInputRef.el.value });
+        }
+
+        /**
+         * @private
+         * @returns {string|undefined}
+         */
+        _computeGuestName() {
+            return this.messaging.currentGuest && this.messaging.currentGuest.name;
+        }
+
+        /**
+         * @private
+         * @returns {boolean}
+         */
+        _computeIsJoinButtonDisabled() {
+            return (
+                typeof this.messaging.currentGuest !== 'undefined' &&
```

Comment:
undefined is tested with `=== undefined` directly (no typeof) but in this case you can just use `this.messaging.currentGuest` as it shouldn't have a falsy value that is not undefined that we would want to exclude